### PR TITLE
ICU-20693 Removing unused code, fixing IDE warnings and preparing for unit tests

### PR DESCRIPTION
**Pull request by hagbard**
This PR fixes IDE warnings and remove any unused code based on running coverage analysis.
In a couple of places methods have been restructured a bit to work better with the upcoming unit tests (review comments added for these).

It also reorders the imports to be consistent with the existing CLDR/ICU standard.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20693
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuData.java [hunk 1]
```diff
@@ -162,4 +164,18 @@ public boolean contains(RbPath rbPath) {
     public boolean isEmpty() {
         return paths.isEmpty();
     }
+
+    @Override public String toString() {
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuData.java]: Just really useful for debugging.

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
     // List of characters to escape in UnicodeSets
     // ('\' followed by any of '\', '[', ']', '{', '}', '-', '&', ':', '^', '=').
     private static final Pattern UNICODESET_ESCAPE =
-        Pattern.compile("\\\\[\\\\\\[\\]\\{\\}\\-&:^=]");
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java]: just some unnecessary \\ quoting that can be removed.

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java [hunk 3]
```diff
@@ -83,20 +83,20 @@ private void writeTo(PrintWriter out, List<String> header) throws IOException {
...
         }
-        closeLastPath(lastPath, 0, out);
+        closeLastPath(0, out);
         out.println();
         out.close();
     }
 
     // Before: Cursor is at the end of the previous line.
     // After: Cursor is positioned immediately after the last closed '}'
-    private void closeLastPath(RbPath lastPath, int minDepth, PrintWriter out) {
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java]: unused variable.

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java [hunk 4]
```diff
@@ -238,9 +238,9 @@ private static boolean mustBeArray(boolean topValues, String name, RbPath rbPath
     }
 
     private void printArray(RbValue rbValue, boolean quote, boolean isSequence, PrintWriter out) {
-        for (int n = 0; n < rbValue.size(); n++) {
+        for (String v : rbValue.getElements()) {
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java]: easier for testing to allow the elements list to be accessed directly (it's immutable anyway).

File: tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/PathMatcher.java [hunk 5]
```diff
@@ -148,12 +148,13 @@ private boolean matchForward(CldrPath path, int matcherIndex) {
     // Make a new, non-interned, unique instance here which we can test by reference to
     // determine if the argument is to be captured (needed as ImmutableMap prohibits null).
     // DO NOT change this code to assign "*" as the value directly, it MUST be a new instance.
+    @SuppressWarnings("StringOperationCanBeSimplified")
```
Comment (hagbard) [inline tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/PathMatcher.java]: Adding warnings where needed to avoid IDE complaints.

**Comment by hagbard** [comment]
Squashed.

**Comment by hagbard** [comment]
Please remember that I cannot merge any of my PRs, so once you're happy with it, please merge.

**Comment by pedberg-icu** [comment]
There are branch conflicts that need resolution before this can be merged....

**Comment by hagbard** [comment]
> There are branch conflicts that need resolution before this can be merged....

Done.

**Pull request merged**